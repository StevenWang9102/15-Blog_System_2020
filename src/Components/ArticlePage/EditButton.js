import React from "react";
import { HashRouter as Router, Link } from "react-router-dom";
import PropTypes from "prop-types";

export const EditButton = props => {

  return (
    <Router>
    <div className='edit-button'>
      <Link to={`/new_post/${props.article_slug}`}>
        <button
          type='button'
          className='btn btn-sm btn-outline-secondary'
          onClick={() => props.onEditArticleClicked(false)}>
          Edit Article
        </button>
      </Link>

      <button
        className='btn btn-sm btn-outline-danger'
        onClick={() => props.onDeleteArticleClicked(props.article_slug)}>
        Delete Article
      </button>
    </div>
    </Router>
  );
};

EditButton.propTypes = {
  article_slug: PropTypes.string,
  onEditArticleClicked: PropTypes.func,
  onDeleteArticleClicked: PropTypes.func,
};