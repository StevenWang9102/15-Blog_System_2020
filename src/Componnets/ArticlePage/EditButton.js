import React from "react";
import { Link } from "react-router-dom";

export const EditButton = props => {
  return (
    <div className='edit-button'>
      <Link to={`/new_post/${props.article_slug}`}>
        <button
          type='button'
          className='btn btn-sm btn-outline-secondary'
          onClick={() => props.onEditArticleClicked(false)}>
          Edit Article
        </button>
      </Link>

      <button
        className='btn btn-sm btn-outline-danger'
        onClick={() => props.onDeleteArticleClicked(props.article_slug)}>
        Delete Article
      </button>
    </div>
  );
};
