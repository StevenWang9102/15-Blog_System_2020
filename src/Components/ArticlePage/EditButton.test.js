import React from "react";
import { EditButton } from "./EditButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <EditButton
        article_slug='test'
        onDeleteArticleClicked={() => {}}
        onEditArticleClicked={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
