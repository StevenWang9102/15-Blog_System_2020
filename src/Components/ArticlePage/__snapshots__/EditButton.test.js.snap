// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="edit-button"
>
  <a
    href="#/new_post/test"
    onClick={[Function]}
  >
    <button
      className="btn btn-sm btn-outline-secondary"
      onClick={[Function]}
      type="button"
    >
      Edit Article
    </button>
  </a>
  <button
    className="btn btn-sm btn-outline-danger"
    onClick={[Function]}
  >
    Delete Article
  </button>
</div>
`;
